Yosemite/Yosemite/Stores/MediaStore.swift

```diff
@@ -34,13 +34,33 @@ public final class MediaStore: Store {
         }
 
         switch action {
+        case .retrieveMediaLibrary(let siteID, let pageFirstIndex, let pageNumber, let pageSize, let onCompletion):
+            retrieveMediaLibrary(siteID: siteID, pageFirstIndex: pageFirstIndex, pageNumber: pageNumber, pageSize: pageSize, onCompletion: onCompletion)
         case .uploadMedia(let siteID, let mediaAsset, let onCompletion):
             uploadMedia(siteID: siteID, mediaAsset: mediaAsset, onCompletion: onCompletion)
         }
     }
 }
 
 private extension MediaStore {
+    func retrieveMediaLibrary(siteID: Int64,
+                              pageFirstIndex: Int,
+                              pageNumber: Int,
+                              pageSize: Int,
+                              onCompletion: @escaping (_ mediaItems: [Media], _ error: Error?) -> Void) {
+        let remote = MediaRemote(network: network)
+        remote.loadMediaLibrary(for: siteID,
+                                pageFirstIndex: pageFirstIndex,
+                                pageNumber: pageNumber,
+                                pageSize: pageSize) { (mediaItems, error) in
+                                    guard let mediaItems = mediaItems, error == nil else {
+                                        onCompletion([], error)
+                                        return
+                                    }
+                                    onCompletion(mediaItems, nil)
```

Comment:
yep this certainly looks more concise! I've also been meaning to write a post about potentially using [Swift's generic enum `Result`](https://developer.apple.com/documentation/swift/result) for callback result type, so that the code looks even more clear for handling success vs. failure scenarios

Reply:
updated in: https://github.com/woocommerce/woocommerce-ios/pull/1900/commits/4b23fbf06dd0f6327bec31feae66e0279160c834

Reply:
> I've also been meaning to write a post about potentially using Swift's generic enum Result for callback result type,

I also briefly looked at that when reviewing this PR. It didn't look like a quick change so I didn't suggest it. Sounds like a good tech debt candidate. 😉

Reply:
I created an issue to try it out :) https://github.com/woocommerce/woocommerce-ios/issues/1926